src/features/applications/request-application-button-commands.test.ts

```diff
@@ -0,0 +1,98 @@
+import { RequestApplication } from "./request-application-button-commands";
+import { setupApplicationTest } from "../../test/helpers/test-setup";
+import { createMockGuild } from "../../test/mocks/create-mock-guild";
+import { createMockButtonInteraction } from "../../test/mocks/create-mock-button-interaction";
+import { asButtonInteraction } from "../../test/utils/discord-conversions";
+import { executeButtonCommand } from "../../test/helpers/execution-helpers";
+import { MessageCreateOptions } from "discord.js";
+
+jest.mock("../../config", () => ({
+  requestDumpThreadId: "111222333",
+}));
+
+describe("RequestApplication", () => {
+  let requestApplication: RequestApplication;
+
+  beforeEach(() => {
+    requestApplication = new RequestApplication();
+  });
+
+  describe("execute", () => {
+    it("should send DM to user and log to request dump channel", async () => {
+      const { interaction, threadChannel } = setupApplicationTest();
+      const resolvedThreadChannel = await threadChannel;
+
+      await requestApplication.execute(asButtonInteraction(interaction));
+
+      expect(interaction.user).toHaveSentDm(/DO NOT REPLY TO THIS MESSAGE/);
+      expect(interaction).toHaveEditedReply(
+        "You have been DM'd the **Volunteer Application**."
+      );
+      expect(resolvedThreadChannel).toHaveSentMessage(
+        `Volunteer Application sent to **${interaction.user.username}** (<@${interaction.user.id}>)`
+      );
+    });
+
+    it("should throw error if request dump channel not found", async () => {
+      const guild = createMockGuild({ threadChannels: [] });
+      const interaction = createMockButtonInteraction({ guild });
+
+      await expect(
+        requestApplication.execute(asButtonInteraction(interaction))
```

Comment:
Seems like the mock generator should already perform asButtonInteraction cast, no?